version.go

```diff
@@ -11,4 +11,16 @@ const ApiVersion = "/go-ipfs/" + CurrentVersionNumber + "/"
 // UserAgent is the libp2p user agent used by go-ipfs.
 //
 // Note: This will end in `/` when no commit is available. This is expected.
-var UserAgent = "go-ipfs/" + CurrentVersionNumber + "/" + CurrentCommit
+func UserAgent(suffix string) string {
+	userAgent := "go-ipfs/" + CurrentVersionNumber + "/"
+	if suffix != "" {
+		userAgent += suffix + "/"
+		// FIXME: An alternative to passing the suffix as string would be to have a
+		//  static `var UserAgentSuffix string` here and modify it during `ipfs daemon`
+		//  startup. The drawback is that ideally this variable should only be allowed
+		//  to be modified there so we instead make it part of the node structure and
+		// explicitly pass it here. Check if this is the correct alternative.
+	}
+	userAgent += CurrentCommit
```

Comment:
nit: please move the optional CurrentCommit before suffix, otherwise the suffix is no longer the suffix :)

Reply:
Thanks for the correction. I misunderstood the `1.19.85` of your example for the `go-ipfs` commit but I see now that it was part of the passed option.